src/main/java/edu/cornell/kfs/fp/batch/service/AccountingDocumentGenerator.java

```diff
@@ -0,0 +1,12 @@
+package edu.cornell.kfs.fp.batch.service;
+
+import java.util.function.Function;
+
+import org.kuali.kfs.krad.document.Document;
+import org.kuali.kfs.sys.document.AccountingDocument;
+
+import edu.cornell.kfs.fp.batch.xml.AccountingXmlDocumentEntry;
+
+public interface AccountingDocumentGenerator<T extends AccountingDocument> {
+    T createDocument(Function<Class<? extends Document>, Document> bareDocumentGenerator, AccountingXmlDocumentEntry documentEntry);

```

Comment:
I'll try renaming them to something like "empty" or "default" instead of "bare", to convey that the Function is generating just the basic document but doesn't populate the rest of the needed data.